### PR TITLE
[#1711] feat(server): Introduce the reconfigurable conf

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?

Introduce the reconfigurable conf for server or other componts.

### Why are the changes needed?

Fix: #1711

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit tests.

### CODE REVIEW

**Comment by zuston** [comment]
I drafted this impl, please take a look. @jerqi

**Comment by zuston** [comment]
> I drafted this impl, please take a look. @jerqi

gentle ping @jerqi

**Comment by zuston** [comment]
> > I drafted this impl, please take a look. @jerqi
> 
> gentle ping @jerqi

ping again

File: common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    private static RssBaseConf rssConf;
+    private ScheduledExecutorService scheduledThreadPoolExecutor;
+
+    private ReconfigurableConfManager(Supplier<RssBaseConf> confSupplier) {
+        rssConf = confSupplier.get();
+        scheduledThreadPoolExecutor = ThreadUtils.getDaemonSingleThreadScheduledExecutor("Refresh-rss-conf");
+        scheduledThreadPoolExecutor.schedule(() -> {
+            try {
+                rssConf.addAll(confSupplier.get());
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java:24]: Maybe you should log more information to display the change of config options.

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java:24]: Yes. This could be logged of changed for better user experience.

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java]: Done

File: common/src/main/java/org/apache/uniffle/common/config/RssConf.java [hunk 2]
```diff
@@ -569,7 +573,7 @@ public <T> RssConf set(ConfigOption<T> option, T value) {
     return this;
   }
 
-  <T> void setValueInternal(String key, T value) {
+  public <T> void setValueInternal(String key, T value) {
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/config/RssConf.java:576]: This method should be used in the class inner from the name. Could we avoid using it?

File: common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            "Update the config option: {} from {} -> {}",
+            configOption.key(),
+            val,
+            rssConf.get(configOption));
+        rssConf.set(configOption, val);
+      }
+    }
+  }
+
+  private boolean isSame(Object v1, Object v2) {
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java:115]: Could we use `Objects.equal`?

Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/ReconfigurableConfManager.java:115]: I don't use this, let me recover.

File: common/src/main/java/org/apache/uniffle/common/config/RssConf.java (hunk 2 above)
Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/config/RssConf.java:576]: I don't use this, let me recover.

File: common/src/main/java/org/apache/uniffle/common/config/RssConf.java [hunk 4]
```diff
@@ -57,6 +57,10 @@ public Set<String> getKeySet() {
     return Sets.newHashSet();
   }
 
+  public Object getObject(String key, Object defaultValue) {
```
[the remark this answers is not in the document]
Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/config/RssConf.java:60]: Used by reconfigurableManager to check whether the prev value is same with the latest value, and then output the log if they are different.

[the remark this answers is not in the document]
Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/config/RssConf.java:60]: I don't see that this method is called this method in any place.

[the remark this answers is not in the document]
Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/config/RssConf.java]: My fault. I forgot to remove this after refactor.

**Review by jerqi — APPROVED** [review]
LGTM.

**Comment by zuston** [comment]
Merged. Thanks @jerqi review

**Pull request merged**